File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.java
class Solution {
    public String reverseWords(String s) {
        // Method 1 Brute Force 

        // String[] arr=s.split(" ");
        // ArrayList<String> list=new ArrayList<>();
        // for(String x:arr){
        //     list.add(reverse(x));
        // }
        // return String.join(" ",list);

        // Method 2 using in-built functions
        String[] arr=s.split(" ");
        StringBuilder sb=new StringBuilder();
        for(int i=0;i<arr.length;i++){
            sb.append(new StringBuilder(arr[i]).reverse());
            if(i<arr.length-1){
                sb.append(" ");
            }
        }
        return sb.toString();

    }
    // public static String reverse(String str){
    //     StringBuilder sb=new StringBuilder(str);
    //     int i=0;
    //     int j=sb.length()-1;
    //     while(i<j){
    //         char temp=sb.charAt(i);
    //         sb.setCharAt(i,sb.charAt(j));
    //         sb.setCharAt(j,temp);
    //         i++;
    //         j--;
    //     }
    //     return sb.toString();
    // }
}

File: 0557-reverse-words-in-a-string-iii/0557-reverse-words-in-a-string-iii.py
class Solution:
    def reverseWords(self, s: str) -> str:
        # Split the string on the basis of spaces
        words=s.split(" ")
        # then using list comprehension built a new list ::-1 is used to
        # reverse list,string or numbers
        rever_words=[word[::-1] for word in words]
        # at lst jon the list with spaces to produce expected output
        return " ".join(rever_words)
        